test/DynamicDependency/data/Microsoft.WindowsAppRuntime.Framework/appxmanifest.xml

```diff
@@ -99,5 +99,17 @@
             <ActivatableClass ActivatableClassId="Microsoft.Windows.Security.AccessControl.SecurityDescriptorHelpers" ThreadingModel="both" />
         </InProcessServer>
     </Extension>
+    <Extension Category="windows.activatableClass.proxyStub">
+        <ProxyStub ClassId="60FC21B2-B396-4D49-94F0-7555869FB93C">
```

Comment:
The proxy/stub CLSIDs for this seem incorrect; did you test that this actually works for both interfaces?

The general rule for proxy/stub CLSIDs is that a given set of IDLs built together will share a single proxy/stub CLSID (1 generated dlldata.c file = 1 proxy/stub CLSID). As a _default_, the proxy/stub CLSID for a dlldata.c file is the IID of the first interface in it. That's why you'll often see registrations where the proxy/stub CLSID matches an IID, which tricks you into making the mistake you've made here, and assuming that they _always_ match.
In addition to being a bit confusing, this default is also unstable; an otherwise reasonable change to the code (e.g. adding a new IDL file to the directory, or adding a new interface to the existing file and putting it above the topmost existing interface for style reasons) could shift which interface is listed first in dlldata.c, resulting in the proxy/stub CLSID changing and breaking all these registrations.

My recommendation is to always explicitly set a proxy/stub CLSID when building a proxy/stub DLL. There are 2 pretty typical patterns:

- Generate a completely new proxy/stub CLSID GUID, and use a C define like "PROXY_CLSID_IS={0x1A7F9F39,0xFB98,0x4405,{0xA6,0x6D,0x95,0x4E,0x07,0x71,0x4E,0x2A}}" to set it. The main downside is that you need to use struct syntax for the GUID, which looks a little gross. The upside is that it's completely self-contained.
- Reference an existing GUID symbol using a C define like "PROXY_CLSID=IID_IWpnForegroundSink". This symbol has to be something that can be resolved when linking the proxy/stub DLL, since it's external from dlldata.c's perspective. You could either use an existing IID (as in my example), or generate a new GUID just for your proxy/stub CLSID and use that.

Whichever approach you go with, your DLL should end up with a single predictable proxy/stub CLSID. Then on the registration side, you'll have a single windows.activatableClass.proxyStub extension, with a single ProxyStub element (with a now-predictable ClassId value), and multiple interfaces underneath it.

Reply:
> Whichever approach you go with, your DLL should end up with a single predictable proxy/stub CLSID.
To expand on this a bit (not relevant to your immediate scenario): you _can_ have a DLL that supports multiple proxy/stub CLSIDs, but only in more complicated scenarios where you're building IDLs in multiple locations. Making this work requires defining different ENTRY_PREFIXes to allow you to combine multiple DllGetClassObject definitions from multiple generated dlldata.c files, and then defining the DLL's _real_ DllGetClassObject manually so that it queries each of the generated prefixed DllGetClassObject functions.

Again, none of this applies here, because this is the simple case: all the IDLs in your proxy/stub DLL are built in a single location. I'm just noting that solutions for the more complicated scenario exist, for future reference.